frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1570,6 +1577,79 @@ impl<T: Config> Pallet<T> {
 	}
 }
 
+#[cfg(any(feature = "try-runtime", test))]
+impl<T: Config> Pallet<T> {
+	fn do_try_state() -> Result<(), &'static str> {
+		Self::try_state_snapshot()?;
+		Self::try_state_signed_submissions_map()?;
+		Self::try_state_phase_off()
+	}
+
+	// [`Snapshot`] state check. Invariants:
+	// - [`DesiredTargets`] exist IFF [`Snapshot`] is present.
+	// - [`SnapshotMetadata`] exist IFF [`Snapshot`] is present.
+	fn try_state_snapshot() -> Result<(), &'static str> {
+		let set = <DesiredTargets<T>>::get().is_some() && <DesiredTargets<T>>::get().is_some();
+
+		match <Snapshot<T>>::take().is_some() {
+            true if !set => Err("If the snapshot exists, the desired targets and snapshot metadata must also exist."),
+            _ => Ok(()),
+		}
+	}
+
+	// [`SignedSubmissionsMap`] state check. Invariants:
+	// - All [`SignedSubmissionIndices`] are present in [`SignedSubmissionsMap`], and no more;
+	// - [`SignedSubmissionNextIndex`] is not present in [`SignedSubmissionsMap`];
```

Comment:
>  Should `SignedSubmissionNextIndex` be equal to the size of `SignedSubmissionsMap`?
Not necessarily, that is the reason why we keep the index in storage (also to avoid iterating over the map).